hidimstat/noise_std.py

```diff
@@ -1,66 +1,259 @@
+
+    method : bool, optional (default='simple')
+        If 'simple', the correlation matrix is estimated by taking the
+        median of the correlation between two consecutive time points
+        and the noise standard deviation at each time point is estimated
+        by taking the median of the standard deviations for every time steps.
+        If 'AR', the order of the AR model is given by `order` and the
+        Yule-Walker method is used to estimate the covariance matrix.
+
+    eps : float, optional (default=1e-2)
+        Length of the cross-validation path.
+        eps=1e-2 means that alpha_min / alpha_max = 1e-2.
+
+    tol : float, optional (default=1e-4)
+        The tolerance for the optimization: if the updates are
+        smaller than ``tol``, the optimization code checks the
+        dual gap for optimality and continues until it is smaller
+        than ``tol``.
+
+    max_iter : int, optional (default=1e+4)
+        The maximum number of iterations.
+
+    n_jobs : int or None, optional (default=1)
+        Number of CPUs to use during the cross validation.
+
+    seed: int, optional (default=0)
+        Seed passed in the KFold object which is used to cross-validate
+        LassoCV. This seed controls the partitioning randomness.
+
+    Returns
+    -------
+    sigma_hat : float
+        Estimated noise standard deviation.
+
+    beta_hat : array, shape (n_features,)
+        Estimated parameter vector.
+    """
+
+    X = np.asarray(X)
+    n_samples, n_features = X.shape
+    n_targets = Y.shape[1]
+
+    if method == 'simple':
+        print('Group reid: simple cov estimation')
+    else:
+        print('Group reid: ' + method + str(order) + ' cov estimation')
+
+    if (max_iter // 5) <= n_features:
+        max_iter = n_features * 5
+        print("'max_iter' has been increased to {}".format(max_iter))
+
+    cv = KFold(n_splits=5, shuffle=True, random_state=seed)
+
+    if fit_Y:
+
+        clf_mtlcv = \
+            MultiTaskLassoCV(eps=eps, normalize=False, fit_intercept=False,
+                             cv=cv, tol=tol, max_iter=max_iter,
+                             n_jobs=n_jobs)
+
+        clf_mtlcv.fit(X, Y)
+        Beta_hat = clf_mtlcv.coef_
+        Error = clf_mtlcv.predict(X) - Y
+        coef_max = np.max(np.abs(Beta_hat))
+        row_max = np.max(np.sum(np.abs(Beta_hat), axis=0))
+
+        if coef_max == 0:
+            support = 0
+        else:
+            support = np.sum(np.sum(np.abs(Beta_hat), axis=0) > tol * row_max)
+
+    else:
+
+        Beta_hat = np.zeros((n_features, n_targets))
+        Error = np.copy(Y)
+        support = 0
+
+    # avoid dividing by 0
+    if support >= n_samples:
+        support = n_samples - 1
+
+    sigma_hat_raw = norm(Error, axis=0) / np.sqrt(n_samples - support)
+
+    if stationary:
+        sigma_hat = np.median(sigma_hat_raw) * np.ones(n_targets)
+        Corr_emp = np.corrcoef(Error.T)
```

Comment:
```suggestion
        corr_emp = np.corrcoef(Error.T)
```

Reply:
`Corr_emp` is a matrix

Reply:
But ok, if I see that it does not matter, capital letters only for `X` and `Y`.